### PR TITLE
#21 Added Altitude, Course, Speed and Accuracy

### LINKED ISSUE #21: Feature request: convert all fields of the Location object, not just latitude and longitude
Hi,

As can be seen in the log fragment below, `LocationUpdatesService` receives a full `Location` object with many fields, but only 'lat' and 'lon' are passed to the Flutter package.
```
I/flutter (28390): locationUpdate: {lat: 51.508281, lon: -0.1397356, logging_enabled: true}
I/flutter (28390): Location Update: Lat: 51.508281 Lon: -0.1397356
D/BackgroundManager(28390): Got success, destroy engine!
D/LocationUpdatesService(28390): New location: Location[fused 51.508503,-0.139252 hAcc=5 et=+1d9h18m55s853ms alt=72.1 vel=11.020249 bear=39.803856 vAcc=1 sAcc=1 bAcc=30 {Bundle[mParcelledData.dataSize=52]}]
```

The conversion seems to be happening in the file below and is hard-coded for only `lat` and `lon`. Would it be possible to modify it to pass all available Location fields to the Flutter package? 

`android\src\main\kotlin\com\icapps\background_location_tracker\flutter\FlutterBackgroundManager.kt`
```
private fun handleInitialized(call: MethodCall, result: MethodChannel.Result, ctx: Context, channel: MethodChannel, location: Location, engine: FlutterEngine) {
        val data = mutableMapOf<String, Any>()
        data["lat"] = location.latitude
        data["lon"] = location.longitude
        data["logging_enabled"] = SharedPrefsUtil.isLoggingEnabled(ctx)
        channel.invokeMethod("onLocationUpdate", data, object : MethodChannel.Result {
            override fun success(result: Any?) {
                Logger.debug("BackgroundManager", "Got success, destroy engine!")
                engine.destroy()
            }

            override fun error(errorCode: String?, errorMessage: String?, errorDetails: Any?) {
                Logger.debug("BackgroundManager", "Got error, destroy engine! $errorCode - $errorMessage : $errorDetails")
                engine.destroy()
            }

            override fun notImplemented() {
                Logger.debug("BackgroundManager", "Got not implemented, destroy engine!")
                engine.destroy()
            }
        })
    }
```

Thank you!

### PR DESCRIPTION
**Pull request by virtualmarc**
Added additional informations to the `BackgroundLocationUpdateData` Object, including Altitude, Course, Speed and Accuracies.

Closes #21

### CODE REVIEW

**Review by NicolaVerbeeck — DISMISSED** [review]
Thanks for your contribution

LGTM

File: test/src/model/background_location_update_data_test.dart [hunk 1]
```diff
@@ -3,14 +3,28 @@ import 'package:flutter_test/flutter_test.dart';
...
+    expect(data.verticalAccuracy, 0.3);
+    expect(data.course, 128.3);
+    expect(data.courseAccuracy, 14.3);
+    expect(data.speed, 12.2);
+    expect(data.speedAccuracy, 0.9);
   });
 
   test('Background location update data', () async {
-    const data = BackgroundLocationUpdateData(lat: 51.45, lon: 4.5);
+    const data = BackgroundLocationUpdateData(lat: 51.45, lon: 4.5, horizontalAccuracy: 1.2, alt: 42.3, verticalAccuracy: 0.3, course: 128.3, courseAccuracy: 14.3, speed: 12.2, speedAccuracy: 0.9);
```
Comment (vanlooverenkoen) [inline test/src/model/background_location_update_data_test.dart:19]: formatting with `,`

File: test/src/model/background_location_update_data_test.dart [hunk 2]
```diff
@@ -3,14 +3,28 @@ import 'package:flutter_test/flutter_test.dart';
 
 void main() {
   test('Background location update data', () async {
-    const data = BackgroundLocationUpdateData(lat: 51.45, lon: 4.5);
+    const data = BackgroundLocationUpdateData(lat: 51.45, lon: 4.5, horizontalAccuracy: 1.2, alt: 42.3, verticalAccuracy: 0.3, course: 128.3, courseAccuracy: 14.3, speed: 12.2, speedAccuracy: 0.9);
```
Comment (vanlooverenkoen) [inline test/src/model/background_location_update_data_test.dart:6]: formatting with `,`

**Review by vanlooverenkoen — CHANGES_REQUESTED** [review]

**Comment by vanlooverenkoen** [comment]
I have requested now tokens on Travis. So normally with your next commit the builds will start again.

**Review by vanlooverenkoen — APPROVED** [review]

**Comment by vanlooverenkoen** [comment]
v1.4.0 is released

**Pull request merged**